server/Server.lua

```diff
@@ -3,10 +3,38 @@
 -----------------------------------------------------------------------------------------------------
 
 RegisterNetEvent("ServerEmoteRequest", function(target, emotename, etype)
+    local ped = GetPlayerPed(source)
+
+    if target == -1 then
+      return
+   end
+   local tped = GetPlayerPed(target)
+   local pedcoord = GetEntityCoords(ped)
+   local targetcoord = GetEntityCoords(tped)
+
+   local distance = #(pedcoord - targetcoord)
+
+   if distance > 2 then
```

Comment:
as AvaN0x mentioned, client side use distance of 3, would be smart to match it
```suggestion
   if distance > 3 then
```